### DISCUSSION TITLE
Panning support

### DISCUSSION THREAD

**Discussion by ASchwad** [reaviz/reaflow#36] [Q&A] [answered]
Hey there,

In the repo code, the Canvas prop `pannable` is activated by default. But for me this does not work in any of the storybook stories. My definition of pannable: Grab the canvas at some spot, where there is no node or edge and drag the mouse from there to move the Canvas into the drag direction.

Do I have a wrong definition or is there a special key that should be pressed to pan the graph?
(e.g. for Zoom it is CTRL + Mousewheel)

Thanks!
Alex

**Comment by amcdnl** [comment, marked as answer]
Panning is done via scrollbars. The panning you are referring to is a "infinite" panning experience typically used for things like maps where the "canvas" is huge. This is the same type of pattern used by most editors type ui's today.

**Reply by ASchwad** [reply]
Ah, that's unfortunate. Any thoughts on how we could integrate infinite panning into reaflow? I would love to have a behaviour as in the [example](https://reactflow.dev/examples/) of Reactflow.

**Reply by amcdnl** [reply]
Actually I chat quite a bit w/ the author of react-flow and they are adding panning like I did here. 

Adding that style of panning is such a nightmare - you have to do a lot of calculations around offset + zoom for dragging. You could add this by turning off panning and then providing your own panning using something like this: https://npm.io/package/panzoom. You might need to tweak some internals.

If you implement this, I'd be happy to add a example but I don't have any plans to add this to the base right now.

**Reply by ASchwad** [reply]
Thanks for your reply. Ultimately I have used the library [react-zoom-pan-pinch](https://github.com/prc5/react-zoom-pan-pinch) and mostly achieved my requirements.

**Reply by amcdnl** [reply]
Thats awesome! You should make a storybook example if possible to show it!

**Reply by himanshu-1034** [reply]
@ASchwad I am also looking for something like this. could you share some code sandbox for reference?

**Reply by himanshu-1034** [reply]
@amcdnl

**Reply by ASchwad** [reply]
@himanshu-1034 I added an example back then to the docs. You can find a description [here](https://reaflow.dev/?path=/story/docs-advanced-plugins--page) and a Storybook example [here](https://reaflow.dev/?path=/story/demos-controls--zoom-external). I hope this helps!

**Reply by himanshu-1034** [reply]
LATEST UPDATE
I was able to add panning using react-zoom-pan-pinch library
Thanks @ASchwad for the doc link. I was missing some styling to make it work.
Latest codesandbox link if anyone else needs help with panning.. 
FIXED CODESANDBOX - https://codesandbox.io/s/reaflow-with-external-panning-usdhdk
